app/timetables/factories.py

```diff
@@ -0,0 +1,161 @@
+import datetime
+
+import factory
+from django.contrib.auth import get_user_model
+from django.utils.text import slugify
+from factory.django import DjangoModelFactory
+
+from . import models
+from common.mixins import TimestampMixin
+
+
+class TimestampFactory(DjangoModelFactory):
+    """TimestampMixin model factory."""
+
+    class Meta:
+        model = TimestampMixin
+        abstract = True
+
+    date_created = datetime.datetime(2008, 1, 1, 0, 0, 0, tzinfo=datetime.timezone.utc)
+    date_modified = datetime.datetime(2009, 1, 1, 0, 0, 0, tzinfo=datetime.timezone.utc)
+
+
+class WeekdayFactory(DjangoModelFactory):
+    """Weekday model factory."""
+
+    class Meta:
+        model = models.Weekday
+
+    name = 'Monday'
+    slug = factory.LazyAttribute(lambda obj: '%s' % slugify(obj.name))
+
```

Comment:
Even so @andela-kerinoso, doing that will cause an additional slugify_field where we will specify the field to be slugified in the factory.